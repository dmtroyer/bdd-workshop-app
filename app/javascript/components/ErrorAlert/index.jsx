import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'reactstrap';

const ErrorAlert = ({ errorMessage }) => (
  <Alert color="danger">Something went wrong - {errorMessage}</Alert>
);
ErrorAlert.propTypes = {
  errorMessage: PropTypes.string.isRequired,
};

export default ErrorAlert;
