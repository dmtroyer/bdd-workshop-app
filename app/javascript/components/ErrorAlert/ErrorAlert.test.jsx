import React from 'react';
import { shallow } from 'enzyme';
import ErrorAlert from '.';

describe('ErrorAlert', () => {
  it('renders correctly in loading state', () => {
    const wrapper = shallow(<ErrorAlert errorMessage="bad thing" />);
    expect(wrapper).toMatchSnapshot();
  });
});
