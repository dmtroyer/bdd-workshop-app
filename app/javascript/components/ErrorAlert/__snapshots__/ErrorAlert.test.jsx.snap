// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorAlert renders correctly in loading state 1`] = `
<Alert
  closeAriaLabel="Close"
  color="danger"
  fade={true}
  isOpen={true}
  tag="div"
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  Something went wrong - 
  bad thing
</Alert>
`;
